cime_config/acme/machines/Makefile

```diff
@@ -185,14 +185,40 @@ ifndef AR
    AR := ar
 endif
 
-ifdef NETCDF_PATH
-  ifndef INC_NETCDF
-    INC_NETCDF:=$(NETCDF_PATH)/include
-  endif
-  ifndef LIB_NETCDF
-    LIB_NETCDF:=$(NETCDF_PATH)/lib
+# To prevent breaking cleaning and building components which don't require NetCDF directly
+ifeq ($(if $(or $(filter pio, $(subst /,  , $(MAKECMDGOALS))), $(filter exec_se, $(subst /,  , $(MAKECMDGOALS)))), true, false), true)
```

Comment:
This detects if we're building either a PIO library or the coupled executable. I'd definitely prefer a simpler method; one thought I had was defining another environment variable for those build steps, but that involves modifying the Python.
The substitution is needed for the pio library because the target is actually a path which depends on the case name; so determining if pio is being built requires searching the string for it.
The substitution isn't actually needed for the coupled executable, so I can remove that.